### ISSUE TITLE
panic: Failed to get slug for key to work out backup location

### ISSUE THREAD

**Issue by zbeekman** [fluidkeys/fluidkeys#197]
On Mac OS 10.14 (Mojave) w/ Xcode 10

Steps:
1. Install and update via your homebrew tap
2. `fk key from-gpg`
3. Found 1 key with fp:1DB1 B5ED E321 22B2 8E56  810D CB21 118C 92A6 4702 <zbeekman@gmail.com>
4. `Connect this key?` Y
5. Issues identified: Usage of bzip2 and key expiration too far in the future
5. `fk key maintain`

```
Fluidkeys found 7 issues for 1DB1 B5ED E321 22B2 8E56  810D CB21 118C 92A6 4702:

 ▸   Primary key expires too far in the future
 ▸   Encryption subkey expires too far in the future
 ▸   Fluidkeys doesn't support BZIP2 compression
 ▸   Fluidkeys doesn't support BZIP2 compression
 ▸   Fluidkeys doesn't support BZIP2 compression
 ▸   Fluidkeys doesn't support BZIP2 compression
 ▸   Fluidkeys doesn't support BZIP2 compression

Fluidkeys will run the following actions:

     [ ] Load private key from gpg
     [ ] Shorten the primary key expiry to 1 Dec 18
     [ ] Set compression preferences to ZLIB, ZIP, Uncompressed
     [ ] Create a new encryption subkey valid until 1 Dec 18
     [ ] Expire the encryption subkey now (0x3538E32D2F3D2C3C)
     [ ] Store updated key in gpg
     [ ] Make backup ZIP file

Make a backup of gpg and run these actions? [Y/n] y

 ▸   Successfully made a backup to:
     /Users/ibeekman/.config/fluidkeys/backups/2018-10-27/gpghome-2018-10-27T13-23-10.tgz

Enter password for 1DB1 B5ED E321 22B2 8E56  810D CB21 118C 92A6 4702:

     [✔] Load private key from gpg
     [✔] Shorten the primary key expiry to 1 Dec 18
     [✔] Set compression preferences to ZLIB, ZIP, Uncompressed
     [✔] Create a new encryption subkey valid until 1 Dec 18
     [✔] Expire the encryption subkey now (0x...)
     [✔] Store updated key in gpg
panic: Failed to get slug for key to work out backup location

goroutine 1 [running]:
github.com/fluidkeys/fluidkeys/backupzip.OutputZipBackupFile(0xc000092240, 0x21, 0xc0000b0690, 0xc000095890, 0xb, 0x0, 0x0, 0x42a05e0, 0xc000082750)
	/var/folders/d0/9mds_bmn77j80rnyw0btcz0r0000gq/T/tmp.Eu0JgH8K/src/github.com/fluidkeys/fluidkeys/backupzip/backupzip.go:41 +0x5c0
main.UpdateBackupZIP.Enact(0xc0000b0690, 0xbeed45a26e937258, 0x3b30177c7, 0x44460e0, 0xc0000a16b8, 0x2e937258, 0x3b30177c7)
	/private/tmp/fluidkeys-20181027-35015-1667c31/fluidkeys-0.2.2/fluidkeys/keymaintain.go:494 +0x62
main.runActions(0xc0000a1680, 0x0, 0x38)
	/private/tmp/fluidkeys-20181027-35015-1667c31/fluidkeys-0.2.2/fluidkeys/keymaintain.go:329 +0xde
main.promptToBackupAndRunActions(0x42a0a60, 0x44637a0, 0xc0000a1680, 0x0, 0xc0000a7500)
	/private/tmp/fluidkeys-20181027-35015-1667c31/fluidkeys-0.2.2/fluidkeys/keymaintain.go:282 +0xf4
main.runKeyMaintain(0xc0000b0690, 0x1, 0x1, 0x42a0a60, 0x44637a0, 0x42a0a40, 0x44637a0, 0xc0000d2d80)
	/private/tmp/fluidkeys-20181027-35015-1667c31/fluidkeys-0.2.2/fluidkeys/keymaintain.go:185 +0x51a
main.keyMaintain(0xc000000000, 0x42644b1)
	/private/tmp/fluidkeys-20181027-35015-1667c31/fluidkeys-0.2.2/fluidkeys/keymaintain.go:43 +0x128
main.keySubcommand(0xc000084f60, 0xc000083860)
	/private/tmp/fluidkeys-20181027-35015-1667c31/fluidkeys-0.2.2/fluidkeys/main.go:184 +0x29f
main.main()
	/private/tmp/fluidkeys-20181027-35015-1667c31/fluidkeys-0.2.2/fluidkeys/main.go:137 +0x19e
```

Furthermore, is there a good reason why fluid keys edited my `gpg.conf` and added `no-tty`? I found that somewhat confusing when trying to inspect my key after the above error was encountered.

**Comment by paulfurley** [comment]
Ouch, thanks for this bug report :\

Regarding `gpg.conf`, we absolutely don't touch that config file. We pass the command-line arguments `gpg2 --pinentry-mode loopback --batch --no-tty`.

Is it possible that `gpg` edited `gpg.conf` itself?!

**Comment by paulfurley** [comment]
This bug is my silly fault for lack of test coverage on the `Email` method (we even have a [trello card](https://trello.com/c/7VeGd2lb/84-do-more-robust-testing-on-pgpkeyemail) for that). I'm working on a fix.

**Comment by zbeekman** [comment]
> Regarding `gpg.conf`, we absolutely don't touch that config file. We pass the command-line arguments `gpg2 --pinentry-mode loopback --batch --no-tty` but we don't mess with config.
> 
> Is it possible that `gpg` edited `gpg.conf` itself?!

Hmmm interesting... I'll investigate. Right now my GPG setup is not super sane: fk + GPG-suite/GPG-tools + kebase.io + gnupg. So my apologies, as I was likely mistaken in thinking fk did that. I also recently upgraded to mojave, and sometimes there are little easter eggs apple throws at you.

**Comment by paulfurley** [comment]
Fixed by #203 and [deployed to Homebrew](https://github.com/fluidkeys/homebrew-tap/commit/5a065abb0a77a9adae72beb065140414e8567b75) - thanks again @zbeekman !
